src/expr.rs

```diff
@@ -2163,6 +2163,12 @@ impl Expression for SimpleExpr {
     }
 }
 
+impl From<SelectStatement> for SimpleExpr {
```

Comment:
What I mean is we remove this `From` and instead have a concrete impl `into_sub_query_expr` somewhere.